keyvi/bin/keyvi_c/c_api.cpp

```diff
@@ -143,20 +143,25 @@ void keyvi_match_destroy(const keyvi_match* match) {
 }
 
 bool keyvi_match_is_empty(const keyvi_match* match) {
-  return match->obj_.IsEmpty();
+  return match->obj_ == nullptr;
 }
 
 double keyvi_match_get_score(const keyvi_match* match) {
-  return match->obj_.GetScore();
+  return match->obj_ == nullptr ? 0 : match->obj_->GetScore();
```

Comment:
do you want to just use `operator bool` here for `match_t`? Doing `match->obj_ ?` will check whether the _stored_ pointer is null, whereas I think (may be mistaken) `match->obj_ == nullptr` implicitly creates a shared_ptr wrapping a nullptr in order to compare `lhs.get() == rhs.get()`.